op-conductor/rpc/backend.go

```diff
@@ -29,22 +29,29 @@ type conductor interface {
 
 // APIBackend is the backend implementation of the API.
 // TODO: (https://github.com/ethereum-optimism/protocol-quest/issues/45) Add metrics tracer here.
-// TODO: (https://github.com/ethereum-optimism/protocol-quest/issues/44) add tests after e2e setup.
 type APIBackend struct {
-	log log.Logger
-	con conductor
+	log            log.Logger
+	con            conductor
+	leaderOverride bool
 }
 
 // NewAPIBackend creates a new APIBackend instance.
 func NewAPIBackend(log log.Logger, con conductor) *APIBackend {
 	return &APIBackend{
-		log: log,
-		con: con,
+		log:            log,
+		con:            con,
+		leaderOverride: false,
```

Comment:
Wouldn't this need to be thread safe? Using an AtomicBool would be enough - otherwise other threads may not see updates to it.

Reply:
The only threads that needs to read this is the main sequencer thread which is used to produce blocks. Currently it doesn't matter if it has a dirty read or not, during the disaster recovery situations, the sequencing will stop, and as long as it can read the override eventually, it should be good to start sequencer.

Initially I thought about adding synchronization here, but elected not to. But since there is concerns around this and to make the code more defensible in the future, changing it to atomic.Bool now.

Reply:
I think eventual consistency is ok here as there is only one write to this by `OverrideLeader` API and all others are reads, it does not quite matter when the reads get the latest value, as long as they get the override and acts accordingly it would be ok for batcher and admin purposes.

Updated to `atomic.Bool` for its better guarantee for reads